src/ec/eddsa.rs

```diff
@@ -22,18 +22,50 @@ pub struct EdDSAParameters;
 
 /// An Ed25519 key pair, for signing.
 pub struct Ed25519KeyPair {
-    private_public: [u8; KEY_PAIR_LEN],
+    private_key: [u8; SEED_LEN],
+
+    // As described in RFC 8032 Section 5.1.6, Step 1, we compute and cache
+    // these values when generating key pairs, since they are used in every
+    // signature. They are defined by:
+    //
+    //     SHA-512(private_key) = private_scalar || private_prefix
+    //
+    private_scalar: [u8; SCALAR_LEN],
+    private_prefix: [u8; PREFIX_LEN],
+
+    public_key: [u8; PUBLIC_KEY_LEN],
 }
 
 /// The raw bytes of the Ed25519 key pair, for serialization.
 pub struct Ed25519KeyPairBytes {
     /// Private key bytes.
     pub private_key: [u8; SEED_LEN],
 
+    private_scalar: [u8; SCALAR_LEN],
+    private_prefix: [u8; PREFIX_LEN],
```

Comment:
We don't need to output these. We just need to output the seed and public key, since that's the standard form of the private key serialization.